features/mbedtls/platform/src/platform_alt.c

```diff
@@ -0,0 +1,59 @@
+ /*
+  *  platform_alt.c
+  *
+  *  Copyright (C) 2018, Arm Limited, All Rights Reserved
+  *  SPDX-License-Identifier: Apache-2.0
+  *
+  *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+  *  not use this file except in compliance with the License.
+  *  You may obtain a copy of the License at
+  *
+  *  http://www.apache.org/licenses/LICENSE-2.0
+  *
+  *  Unless required by applicable law or agreed to in writing, software
+  *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+  *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  *  See the License for the specific language governing permissions and
+  *  limitations under the License.
+  *
+  */
+
+#include "mbedtls/platform.h"
+#if defined(MBEDTLS_PLATFORM_SETUP_TEARDOWN_ALT)
+
+static int reference_count = 0;
+
+int mbedtls_platform_setup( mbedtls_platform_context *ctx )
+{
+    int ret = 0;
+    if( ctx == NULL )
+        return ( -1 );
```

Comment:
I don't think we should be returning -1, however, we have failed to define in Mbed TLS any error codes for function.

Maybe we should define error codes now locally in Mbed OS, and upstream them to Mbed TLS in time for the following release of Mbed OS? Possibly `INVALID_INPUT` for the case above, and `HW_FAILURE` in case of failure from the accelerator. What do you think?

Reply:
I agree, but as you said, there is currently no error codes available

Reply:
You could define some locally in Mbed OS, and also submit a PR to upstream the same codes. 

As and when the version of Mbed TLS with the new codes becomes available, we can remove the local definitions. 

We control both sides (upstream and downstream) so there's no risk of reusing the same values.

Reply:
I created https://github.com/ARMmbed/mbedtls/pull/1993 to add the platform error code